query_test.go

```diff
@@ -3220,6 +3220,28 @@ FROM
 				{"false", "boolean"},
 			},
 		},
+
+		// subquery expr
+		{
+			name:         "subquery_expr_type_scalar",
+			query:        "SELECT (SELECT 1)",
```

Comment:
It does not matter whether the subquery is in the SELECT/WHERE/HAVING clause etc., the scalar type seems to be the problem.

Would you like to add these test cases?

```sql
SELECT * FROM UNNEST([1, 2, 3]) AS val WHERE val = (SELECT 1)
SELECT * FROM UNNEST([1, 2, 3]) AS val GROUP BY val HAVING val = (SELECT 1)
SELECT ABS((SELECT 1))
```

Reply:
Please add them !

Reply:
added.